tuplex/core/src/ee/local/LocalBackend.cc

```diff
@@ -139,12 +137,10 @@ namespace tuplex {
         } else
             throw std::runtime_error("unknown stage encountered in local backend!");
 
-        // detach from driver
-        _driver->setHistoryServer(nullptr);
-
         // send final message to history server to signal job ended
-        if(_historyServer) {
+        if(_historyServer && stage->predecessors().size() == stage->plan()->getNumStages() - 1) {
```

Comment:
what does the second part of the if condition check? Can we add a comment?

Reply:
The second part of the condition checks that the job is actually finished (all stages have been iterated through). Comment added.